include/boost/geometry/algorithms/detail/envelope/box.hpp

```diff
@@ -40,36 +42,124 @@ namespace detail { namespace envelope
+    std::size_t Dimension,
+    std::size_t DimensionCount,
+    typename CS_Tag
+>
+struct envelope_indexed_box
 {
-    template<typename BoxIn, typename BoxOut>
+    template <typename BoxIn, typename BoxOut>
     static inline void apply(BoxIn const& box_in, BoxOut& mbr)
     {
-        BoxIn box_in_normalized = detail::return_normalized<BoxIn>(box_in);
+        detail::indexed_point_view<BoxIn const, Index> box_in_corner(box_in);
+        detail::indexed_point_view<BoxOut, Index> mbr_corner(mbr);
+
+        detail::conversion::point_to_point
+            <
+                detail::indexed_point_view<BoxIn const, Index>,
+                detail::indexed_point_view<BoxOut, Index>,
+                Dimension,
+                DimensionCount
+            >::apply(box_in_corner, mbr_corner);
+    }
+};
 
-        geometry::transform(box_in_normalized, mbr);
+template
+<
+    std::size_t Index,
+    std::size_t DimensionCount
+>
+struct envelope_indexed_box<Index, 0, DimensionCount, spherical_equatorial_tag>
+{
+    template <typename BoxIn, typename BoxOut>
+    static inline void apply(BoxIn const& box_in, BoxOut& mbr)
+    {
+
+       // transform() does not work with boxes of dimension higher
+        // than 2; to account for such boxes we transform the min/max
+        // points of the boxes using the indexed_point_view
+        detail::indexed_point_view<BoxIn const, Index> box_in_corner(box_in);
+        detail::indexed_point_view<BoxOut, Index> mbr_corner(mbr);
+
+        // first transform the units
+        transform_units(box_in_corner, mbr_corner);
+
+        // now transform the remaining coordinates
+        detail::conversion::point_to_point
+            <
+                detail::indexed_point_view<BoxIn const, Index>,
+                detail::indexed_point_view<BoxOut, Index>,
+                2,
+                DimensionCount
+            >::apply(box_in_corner, mbr_corner);
     }
 };
 
+template
+<
+    std::size_t Index,
+    std::size_t Dimension,
+    std::size_t DimensionCount
+>
+struct envelope_indexed_box<Index, Dimension, DimensionCount, geographic_tag>
+    : envelope_indexed_box
+        <
+            Index, Dimension, DimensionCount, spherical_equatorial_tag
+        >
+{};
+
 
-template <typename CSTag>
+template
+<
+    typename std::size_t Dimension,
+    std::size_t DimensionCount,
+    typename CS_Tag
+>
 struct envelope_box
 {
     template<typename BoxIn, typename BoxOut>
     static inline void apply(BoxIn const& box_in, BoxOut& mbr)
     {
-        geometry::convert(box_in, mbr);
+        envelope_indexed_box
+            <
+                min_corner, Dimension, DimensionCount, CS_Tag
+            >::apply(box_in, mbr);
+
+        envelope_indexed_box
+            <
+                max_corner, Dimension, DimensionCount, CS_Tag
+            >::apply(box_in, mbr);
+    }
+};
+
+
+template <typename std::size_t DimensionCount>
+struct envelope_box<0, DimensionCount, spherical_equatorial_tag>
```

Comment:
You could certainly call `envelope_box<2, 3, geographic_tag>` it would simply convert the coordinate of dimension 2.
If I understand your question correctly, the reason for dispatch using dimensions as well is due to the new general dispatch design for envelope, that also takes a dimension range into account. I agree that it might make less sense for `envelope_box`, but on the other hand this is the general dispatch design, and I thought it would be better to follow it globally, for uniformity.